tla-io/src/main/scala/at/forsyte/apalache/io/lir/NameReplacementMap.scala

```diff
@@ -0,0 +1,15 @@
+package at.forsyte.apalache.io.lir
+
+import scala.collection.mutable.HashMap
+
+/**
+ * This class produces counterexamples in the Informal Trace Format.
+ *
+ * @param writer
+ *   a print writer to use
+ * @author
+ *   Philip Offtermatt
+ */
+package object NameReplacementMap {
```

Comment:
Do we assume the PrettyWriter to be a singleton?
Otherwise, we might need multiple `NameReplacementMap`s?
In short, I'm not sure this should be a `package object` rather than simply a `class`.

Reply:
I updated the comment, does this make it clearer? It's supposed to be globally accessible, since I use it to propagate information from the temporal encoding pass to the counterexample writer at the very end. It also doesn't make sense to have multiple NameReplacementMaps (in what I'm using it for) - if there are multiple PrettyWriters, they should all access the same one

Reply:
I think Shon had a similar [comment below](https://github.com/informalsystems/apalache/pull/1823#discussion_r902699251), so I will follow up there.